packages/core/src/plugins/rsdoctor.ts

```diff
@@ -45,9 +48,14 @@ export const pluginRsdoctor = (): RsbuildPlugin => ({
 
       let isAutoRegister = false;
 
+      function isRsdoctorPlugin(plugin: Configuration['plugins'] & { isRsdoctorPlugin?: boolean }) {
+        return plugin != null && typeof plugin.isRsdoctorPlugin === 'boolean';
```

Comment:
`!=` is not recommended to use

```suggestion
        return plugin?.isRsdoctorPlugin === true;
```